cpp/harvest/core.cc

```diff
@@ -54,10 +56,12 @@ namespace {
         "\t- input_dir: A dir with multiple JSON files to merge, typically from a search result.\n"
         "\t- output_file: The directory to store the merged JSON result file.\n"
         "\n"
-        "filter input_file output_file filtered_file\n"
+        "filter input_file output_file filtered_file keyword_file [data_provider_filter_type] [data_provider_ids_file]\n"
         "\t- input_file: A single JSON input file.\n"
         "\t- output_file: The target JSON file without filtered datasets.\n"
         "\t- filtered_file: File to store datasets that have been removed when filtering. The reason will be stored in each JSON entry.\n"
+        "\t- data_provider_filter_type: 'equal' or 'not_equal'.\n"
```

Comment:
"equal" and "not_equal" are not really technically correct because we compare against a list, so we should rather use terms like "whitelist" and "blacklist" or "contains" and "missing" and so on.